### PR TITLE
Python: finalises config feature

### PR DESCRIPTION
**Pull request by akuesters**
- Removed `config.hpp`
- Removed `mpi_compiled()` and `mpi4py_compiled()` 
- Replaced them with `config.mpi` and `config.mpi4py` in unit tests

### CODE REVIEW

File: python/config.cpp [hunk 1]
```diff
@@ -1,9 +1,50 @@
...
+#include <iomanip>
+#include <ios>
 
-#include "config.hpp"
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
 
 namespace pyarb {
+    
+// Create and return a dictionary
```
Comment (bcumming) [inline python/config.cpp]: Say what the dictionary is, i.e. something like "returns a python dictionary that python users can use to look up which options the Arbor library was configured with at compile time."

File: python/config.cpp [hunk 2]
```diff
@@ -1,9 +1,50 @@
...
+#else
+    dict[pybind11::str("gpu")]     = pybind11::bool_(false);
+#endif
+    dict[pybind11::str("version")] = pybind11::str(ARB_VERSION);
+    return dict;
+}
+
+void print_config(const pybind11::dict &d) {
+    std::stringstream s;
+    s << "Arbor's configuration:" << std::endl;
```
Comment (bcumming) [inline python/config.cpp]: just `<< Arbor's configuration:\n";
`std::endl` can have heavy performance cost, and should only be used when you really want to force that the output is flushed.

File: python/test/unit_distributed/test_contexts_arbmpi.py [hunk 3]
```diff
@@ -15,10 +16,15 @@
 except ModuleNotFoundError:
     from test import options
 
+# check Arbor's configuration of mpi
```
Comment (bcumming) [inline python/test/unit_distributed/test_contexts_arbmpi.py]: Remove the named tuple tests. We were interested in named tuples, if we could get our `arbor.config()` to return one, but we have settled on dictionaries.

File: python/test/unit_distributed/test_contexts_arbmpi.py [hunk 4]
```diff
@@ -36,7 +42,7 @@ def setUpClass(self):
     def tearDownClass(self):
         #print("tearDown --- TestContextMPI class")
```
Comment (bcumming) [inline python/test/unit_distributed/test_contexts_arbmpi.py]: Remove this commented out code.

File: python/test/unit_distributed/test_contexts_mpi4py.py [hunk 5]
```diff
@@ -3,6 +3,7 @@
 # test_contexts_mpi4py.py
 
 import unittest
+from collections import namedtuple
```
Comment (bcumming) [inline python/test/unit_distributed/test_contexts_mpi4py.py]: Again, remove all `namedtuple` mentions.

File: python/config.cpp (hunk 1 above)
Comment (akuesters) [reply python/config.cpp]: Done

File: python/config.cpp (hunk 2 above)
Comment (akuesters) [reply python/config.cpp]: Done

File: python/test/unit_distributed/test_contexts_arbmpi.py (hunk 3 above)
Comment (akuesters) [reply python/test/unit_distributed/test_contexts_arbmpi.py]: Done

File: python/test/unit_distributed/test_contexts_arbmpi.py (hunk 4 above)
Comment (akuesters) [reply python/test/unit_distributed/test_contexts_arbmpi.py]: Done

File: python/test/unit_distributed/test_contexts_mpi4py.py (hunk 5 above)
Comment (akuesters) [reply python/test/unit_distributed/test_contexts_mpi4py.py]: Done

**Pull request merged**